tests/test_flow.py

```diff
@@ -4969,3 +4969,49 @@ def test_parallel_execution_with_large_data():
         else:
             for expected_gpu, runnable_result in enumerate(result.values()):
                 assert runnable_result == {"data_size": data_size, "n": n, "gpu": expected_gpu}
+
+
+class RunnableShared(ParallelExecutionRunnable):
+    execution_mechanism = ParallelExecutionMechanisms.shared_executor
+
+
+def test_parallel_execution_with_shared():
+    busy_wait_pool = RunnableBusyWait("busy1")
+    busy_wait_dedicated = RunnableBusyWait("busy2")
+    busy_wait_dedicated.execution_mechanism = "dedicated_process"
+
+    runnables = [RunnableShared("busy2"), busy_wait_pool, RunnableShared("thread1")]
+
+    class MyParallelExecution(ParallelExecution):
+        def select_runnables(self, event):
+            return None
+
+    class MyContext:
+        def __init__(self, executor: RunnableExecutor):
+            self.executor = executor
+
+    my_executor = RunnableExecutor()
+    my_executor.add_runnable(busy_wait_dedicated)
+    my_executor.add_runnable(RunnableSleep("thread1"))
+    my_context = MyContext(executor=my_executor)
+
+    parallel_execution = MyParallelExecution(runnables, context=my_context)
+    reduce = Reduce([], lambda acc, x: acc + [x])
+
+    source = SyncEmitSource()
+    source.to(parallel_execution).to(reduce)
+
+    start = time.monotonic()
+    controller = source.run()
+    controller.emit(0)
+    controller.terminate()
+    termination_result = controller.await_termination()
+    end = time.monotonic()
+
+    assert end - start < 6
```

Comment:
Pretty sure in this test we want to make sure it's less than 3. Because they could be running in serial and still be done in less than 6 seconds.